core/pkg/artifacts/linker.go

```diff
@@ -72,3 +81,41 @@ func (al *ArtifactLinker) Link() error {
 
 	return err
 }
+
+func (al *ArtifactLinker) resolveOrgEntityName(portfolioEntity string, organization string) (string, error) {
+	// Fetches the org entity of the portfolio entity to
+	// 1. validate the user inputted the correct display org name or org entity name and
+	// 2. return the org entity name so we can use the correct entity name to link the artifact.
+	orgFieldNames, err := GetGraphQLFields(al.Ctx, al.GraphqlClient, "Organization")
```

Comment:
Are we doing this every time the user invokes `link_artifact`? Is that reasonable?

Reply:
link artifact does not happen that often so its reasonable but not ideal. 
In the internal API in python we just make this check once and save the value to the internal api class. Ideally we can have something similar to that here. Or have a global server version check etc.
We(Me, Tony, and Katia?) have had discussions on how to check backwards compatibility for artifacts fields and I believe this is what we landed on for now.

Reply:
+1, may be good to figure out how to cache (maybe with ttl) the introspection requests.  That seems out of scope for this PR, but would mirror what's currently done in the Python API 

We do need a way to check compatibility against older server versions, and so far introspection seems like the most reliable / least brittle way to do it